File: backend/voting-app/src/main/java/com/tu/votingapp/entities/surveys/SurveyResponseEntity.java
package com.tu.votingapp.entities.surveys;

import com.tu.votingapp.entities.UserEntity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(
        name = "survey_responses",
        uniqueConstraints = @UniqueConstraint(columnNames = {"user_id", "question_id"})
)
@Data
@NoArgsConstructor
@AllArgsConstructor
public class SurveyResponseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /** Who answered */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private UserEntity user;

    /** Which survey question */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "question_id", nullable = false)
    private SurveyQuestionsEntity question;

    /** Which option they picked */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "option_id", nullable = false)
    private SurveyOptionEntity option;

    /** When they submitted their answer */
    @Column(name = "responded_at", nullable = false)
    private LocalDateTime respondedAt;
}

File: frontend/android/app/src/main/java/com/example/android/entity/survey/SurveyResponseEntity.kt
package com.example.android.entity.survey

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "survey_responses")
data class SurveyResponseEntity(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    val id: Long? = null,

    @ColumnInfo(name = "user_id")
    val userId: Long,

    @ColumnInfo(name = "question_id")
    val questionId: Long,

    @ColumnInfo(name = "option_id")
    val optionId: Long,

    @ColumnInfo(name = "responded_at")
    val respondedAt: String // Store as ISO-8601 formatted String (e.g., "2024-04-18T10:15:30")
)